src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,233 @@
+		commandName = "/usr/bin/open"
+		args = []string{"-a", applicationPath}
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func moveToParent(fullPath string, numberTimes int) string {
+	fullPath = path.Clean(fullPath)
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+func getWindowsRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	localAppDataGetters := []func() string{
+		func() string {
+			return os.Getenv("LOCALAPPDATA")
+		},
+		func() string {
+			homeDrive := os.Getenv("HOMEDRIVE")
+			if homeDrive == "" {
+				return ""
+			}
+			homePathPart := os.Getenv("HOMEPATH")
+			if homePathPart == "" {
+				return ""
+			}
+			return path.Join(homeDrive+homePathPart, "AppData", "Local")
+		},
+		func() string {
+			homePath := os.Getenv("HOME")
+			if homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+		func() string {
+			homePath, err := os.UserHomeDir()
+			if err != nil || homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+	}
+	for _, localAppDataGetter := range localAppDataGetters {
+		localAppDataDir := localAppDataGetter()
+		candidatePath := checkExistence(path.Join(localAppDataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return ""
+}
+
+func getMacOSRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		// we're at .../Applications/R D.app/Contents/Resources/resources/darwin/bin
+		// and want to move to /Applications (or ~/Applications, or something else)
+		normalParentPath := moveToParent(rdctlPath, 7)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.app"), 0)
+		if candidatePath != "" {
+			// No need to verify permissions on the directory App/R D.app -- it must be a directory, or it
+			// couldn't be the ancestor parent of rdctl. And it must be executable/readable enough for us
+			// to have been able to path.Dir() our way up to it.
+			return candidatePath
+		}
+	}
+	return checkExistence(path.Join("/Applications", "Rancher Desktop.app"), 0)
+}
+
+func getLinuxRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
```

Comment:
Presumably this goes from `…/rancher-desktop/resources/linux/bin/rdctl`? Why is it going up 5 steps (when 4 gets you to the app dir)?

Reply:
You missed one of the resources -- for me the full path is `~/opt/rancher-desktop-1.2.0-512-gebedf8a2-linux/resource/resources/linux/bin/rdctl` so you need 5 `dirname`s (in bash) to get to the directory that contains the executable called `rancher-desktop`.